### PR TITLE
Prev next fix

### PR DESCRIPTION
**Pull request by antoine-de**
some list param were missing in links

small refactoring  of the links

this refactoring should be more complete, but we cannot do this before
the v2 and some thought about the internal link handling

links example:
api endpoint

``` json
"versions": [
    {
        "description": "Current version of the api",
        "links": [
            {
                "href": "http://localhost:5000/v1/",
                "rel": "api",
                "templated": false,
                "type": "api"
            }
        ],
        "status": "current",
        "value": "v1"
    }
]
```

call on 'v1'

``` json
{

    "links": [
        {
            "description": "Coverage of navitia",
            "href": "http://localhost:5000/v1/coverage/",
            "rel": "coverage",
            "templated": false,
            "type": "coverage"
        },
        {
            "description": "Inverted geocoding for a given coordinate",
            "href": "http://localhost:5000/v1/coord/0.0%3B0.0",
            "rel": "coord",
            "templated": true,
            "type": "coord"
        },
        {
            "description": "Compute journeys",
            "href": "http://localhost:5000/v1/journeys",
            "rel": "journeys",
            "templated": false,
            "type": "journeys"
        }
    ]

}
```

some links on journeys:

``` json
links": [
    {
        "href": "http://localhost:5000/v1/journeys?to=poi%3ATAOGen861&min_nb_journeys=3&from=stop_area%3ASCF%3ASA%3ASAOCE87543009&datetime_represents=arrival&datetime=20141030T153116",
        "rel": "prev",
        "templated": false,
        "type": "prev"
    },
    {
        "href": "http://localhost:5000/v1/coverage/stop_points/{stop_point.id}?collection=stop_points",
        "rel": "stop_points",
        "templated": true,
        "type": "stop_point"
    },
```

Note: for the 'next' links the type should be 'journeys' but since it was 'next' before we cannot change that before the v2 :cry:

### CODE REVIEW

**Comment by antoine-de** [comment]
fix http://jira.canaltp.fr/browse/NAVITIAII-1224

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py [hunk 1]
```diff
@@ -30,11 +30,61 @@
...
+    :param url: url forwarded to flask's url_for
+    :param rel: relation of the link to the current object
+    :param _type: type of linked object
+    :param templated: if the link is templated ({} is the url)
+    :param description: description of the link
+    :param kwargs: args forwarded to url_for
+    :return: a dict representing a link
+    """
+    #if no type, type is rel
+    if not _type:
```
Comment (l-vincent-l) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py:50]: I'm not sure we want that. It's should be rare.

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py [hunk 2]
```diff
@@ -30,11 +30,61 @@
...
+        _type = rel
+
+    d = {
+        "href": url_for(url, _external=True, **kwargs),
+        "templated": templated,
+        "rel": rel,
+        "type": _type
+    }
+    if description:
+        d['description'] = description
```
Comment (l-vincent-l) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py]: Actually we saw HAL is using the keyword title.

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py [hunk 3]
```diff
@@ -30,11 +30,61 @@
...
+def create_internal_link(rel, _type, id, templated=False, description=None):
+    """
+    :param rel: relation of the link to the current object
+    :param _type: type of linked object
+    :param id: id of the link
+    :param templated: if the link is templated ({} is the url)
+    :return: a dict representing a link
+    """
+    #if no type, type is rel
+    if not _type:
```
Comment (l-vincent-l) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py:74]: Do we want that?

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py [hunk 4]
```diff
@@ -30,11 +30,61 @@
...
+        _type = rel
+
+    d = {
+        "templated": templated,
+        "rel": rel,
+        "type": _type
+    }
+    if description:
+        d['description'] = description
```
Comment (l-vincent-l) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py]: title?

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py (hunk 1 above)
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py:50]: nop it's quite common in our code (but you're right it should not)

**Pull request merged**